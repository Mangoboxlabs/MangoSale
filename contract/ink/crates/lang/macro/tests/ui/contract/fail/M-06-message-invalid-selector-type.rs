use ink_lang as ink;

#[ink::contract]
mod message_invalid_selector {
    #[ink(storage)]
    pub struct MessageInvalidSelector {}

    impl MessageInvalidSelector {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message, selector = true)]
        pub fn invalid_selector(&self) {}
    }
}

fn main() {}
