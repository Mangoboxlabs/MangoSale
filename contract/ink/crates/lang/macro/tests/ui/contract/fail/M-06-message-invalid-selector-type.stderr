error: expecteded 4-digit hexcode for `selector` argument, e.g. #[ink(selector = 0xC0FEBABE]
  --> $DIR/M-06-message-invalid-selector-type.rs:14:24
   |
14 |         #[ink(message, selector = true)]
   |                        ^^^^^^^^^^^^^^^
